### DISCUSSION TITLE
What's the advantage of using this plugin?

### DISCUSSION THREAD

**Discussion by x-xyzxyz** [stayallive/wp-sentry#163] [Q&A] [answered]
Hello, and thank you so much for the huge amount of work you put into this plugin!

I was wondering, besides tracking/identifying logged in users, what are the advantages of using this plugin VS using the PHP+JS SDKs?

I'm asking because for me it's hard to get bits of config/filters from Sentry's documentation and convert them into a format the plugin can digest (through WP hooks etc), so I was wondering what I'd lose if I switched to normal PHP+JS integration without the plugin :)

Thank you,
Alessandro

**Comment by stayallive** [comment]
Hi Alessandro, I'm assuming you already looked through the code so you know exactly what you'll miss :)

Nothing much, this is a thin wrapper around the PHP/JS SDK's which does some auto-configuration to make the installation as simple as plugin installation and setting the DSN in the settings. In addition some extra configuration helpers to configure the SDK's more easily "the WordPress way".

Not sure if that answers the question but there is nog magic WordPress things going on that you won't be able to replicate without the plugin 😉 

Having said that, I'm very open to ideas and improvements to help ease the problems you are seeing and why you are considering not using the plugin at all, maybe there is room for improvement here that I'm missing and you can help me identify, so if you have the time I'd appreciate some of your insights on what you are trying to do and why it's not easy or convienient for your use case!

But no hard feelings if you decide to implement your own ways of using the PHP/JS SDK's, if you are at that point or need that much configuration it might be best to have all the control!

**Reply by x-xyzxyz** [reply]
Hey there,
Yes, this is the kind of answer I was hoping for, thank you!

Your plugin is great, I think it's just me not being the smartest when it comes to this kind of things 😄 

The things I'm struggling with are mainly these 2:

1. Setting up things through `wp-config.php` is a bit painful because I only have the theme in Git's version control, and with 3 environments (dev, staging, production), it's hard to replicate (and easy to forget) these changes across all 3 environments whenever I make a change. It would be great to be able to define configs and settings in `functions.php`, for example.
2. The biggest problem I'm having is to convert the config examples from Sentry's docs into a format that your plugin can digest.
For example, converting things like this:
```js
beforeSend(event, hint) {
    if (event.request.headers["User-Agent"] === "WP Rocket / Preload") {
      console.log("Dropping error event due to test user-agent")
      return null;
    }
    return event; 
  },
```
into a format like this (from your [docs](https://github.com/stayallive/wp-sentry#client-side-hook)):

```js
add_action( 'wp_enqueue_scripts', function () {
	wp_add_inline_script( 'wp-sentry-browser', 'function wp_sentry_hook(options) { return someCheckInYourCode() ? true : false; }', 'before' );
} );
```

I've asked ChatGPT and it fed me something like this, though I'm not sure it's actually correct, and I find JS within PHP to be cumbersome and hard to read and maintain:

```php
add_action( 'wp_enqueue_scripts', function () {
    wp_add_inline_script( 'wp-sentry-browser', '
        function wp_sentry_hook(options) { 
            options.beforeSend = function(event, hint) {
                if (event.request && event.request.headers && event.request.headers["User-Agent"] === "WP Rocket / Preload") {
                    console.log("Dropping error event due to test user-agent");
                    return null;
                }
                return event; 
            };
            return options;
        }', 'before' );
} );


```

So yeah overall the main reason why I was thinking about not using the plugin, is that Sentry's docs will always give me pieces of info that won't work with the plugin out of the box, and it's hard to come up with a way to convert them each time, especially as a first time Sentry user.

Hope this answer your questions, and just to be clear there's nothing wrong with your plugin!

**Reply by stayallive** [reply, marked as answer]
Ah, yes the JavaScript options especially are a PITA and I don't have a good way to fix that so especially for the browser SDK it indeed might be better to include your own integration to solve that. But surprisingly the ChatGPT answer looks pretty decent but haven't tested it.

Eventually it might be worth keeping the plugin around for the PHP integration since I do plan on having performance tracing integrated which would be a lot of work to do yourself, but that is not yet implemented but maybe the prospect of that is worth of keeping the PHP side of this plugin around. The browser version taking in-house makes total sense to me and is something I have suggested to numerous people in the past that needed a little extra configuration than I can provide through PHP.

For the `wp-config.php` setting you could add a file in your theme you require from the `wp-config.php` to sync the settings a little better but since Sentry excells when loaded as early as possible that is always going to be an interesting problem to solve.

This is also not a official plugin and I can't keep up with every single option and making sure I have an example for every example in the official docs unfortunately so unless Sentry decides WordPress is worth the effort to bring in-house we'll have to do with some community effort to cobble together enough information.

**Reply by x-xyzxyz** [reply]
hello!
thank you for your reply, I think the performance tracing alone is enough to convince me to stick to your plugin, looking forward to that!

I guess I won't have to change Sentry's config too often once I've properly set it up, and I don't have a lot of things I wanna change anyway, so I'll push through these few settings and stick to the WP plugin for the time being :)

Very odd Sentry won't provide an official plugin given how many WP websites are there. Even Rollbar provide one and they look like a much smaller company.

Anyway thanks a lot for all these walls of text and for doing God's work by building and maintaining this plugin!

Take care,
Alessandro
